### DISCUSSION TITLE
Using soft_predict

### DISCUSSION THREAD

**Discussion by JPico6** [omadson/fuzzy-c-means#45] [Q&A]
Greetings. Can you offer guidance on properly using the soft_predict method? I apologize if I missed something obvious. The output preds is always just the base rates, so in the case of 3 clusters, .333, .333, .333

X = np.array of dimension 80,000 x 140

```
from fcmeans import FCM

fcm = FCM(n_clusters=3, m=1.5, max_iter=100, random_state=1)
fcm.fit(X)
# note I've experimented with different params

pred = fcm.soft_predict(X)

print(pred)
```

Out[60]: array([[0.33333334, 0.33333333, 0.33333333],
       [0.33333333, 0.33333333, 0.33333333],
       [0.33333334, 0.33333333, 0.33333333],
       ...,
       [0.33333334, 0.33333333, 0.33333332],
       [0.33333333, 0.33333333, 0.33333333],
       [0.33333334, 0.33333333, 0.33333333]])

**Comment by omadson** [comment]
Hi @JPico6. I hope it can still help you. Well I created a [simple example](https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/75ab1750f37b27db8e135c2c4f2139da6b641609?urlpath=lab%2Ftree%2Fdemo%2Fsoft_prediction_example.ipynb) with the `soft_prediction` method and it works fine. I really don't understand why it doesn't work with your data. Could you give more information about your data?
